3p/frame-metadata.js

```diff
@@ -0,0 +1,166 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {dev} from '../src/log';
```

Comment:
i think it's a good idea to extract this part out as a separate file. but since you're getting that far, does it make sense to make it a class?
```js
class FrameMetaData {
  constructor() {
     // pasring iframe.name
  }
  getAttributeData() {..}
    // getContextState();
}
```